### PR TITLE
Fix e2e tests with userID encoding change

### PR DESCRIPTION
**Pull request by sbryzak**
Fixes e2e tests to work with new UserID encoding rules.

https://github.com/codeready-toolchain/member-operator/pull/342

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -451,4 +451,8 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("deactivate UserSignup and ensure that all user and identity resources are deleted", func(t *testing.T) {
+		// TODO
+	})
+
```
Comment (MatousJobanek) [inline test/e2e/e2e_test.go:457]: This is TODO for the other PR, correct? https://github.com/codeready-toolchain/member-operator/pull/341
```suggestion
```

File: testsupport/user_assertions.go [hunk 2]
```diff
@@ -79,12 +74,9 @@ func VerifyResourcesProvisionedForSignup(t *testing.T, awaitilities wait.Awaitil
 		assert.NoError(t, err)
 
 		// Verify provisioned Identity
-		userID := userAccount.Spec.UserID
-		if !dns1123ValueRegexp.MatchString(userAccount.Spec.UserID) {
-			userID = fmt.Sprintf("b64:%s", base64.RawStdEncoding.EncodeToString([]byte(userAccount.Spec.UserID)))
-		}
+		identityName := identity.NewIdentityNamingStandard(userAccount.Spec.UserID, "rhd").IdentityName()
```
Comment (MatousJobanek) [inline testsupport/user_assertions.go:77]: I would expect the same logic as we use for creating the identities:
```go
userID := userAccount.Spec.UserID
if strings.ContainsAny(userAccount.Spec.UserID, ":/") {
	userID = fmt.Sprintf("b64:%s", base64.RawStdEncoding.EncodeToString([]byte(userAccount.Spec.UserID)))
}
```

Comment (MatousJobanek) [reply testsupport/user_assertions.go:77]: Ah, you reuse the logic from toolchain-common - didn't notice that, sorry :man_facepalming:

**Review by MatousJobanek — APPROVED** [review]

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**